docs/interfaces.md

```diff
@@ -0,0 +1,431 @@
+# FPC Component Interfaces and States
+
+This document defines the interfaces exposed by the FPC components and their internal state.
+
+
+# ERCC
+
+## Interface:
+The ERCC interface is implemented through a "normal" chaincode interface. All functions can be called by admin or via cc2cc.
+
+```go
+// chaincode interface (exposed to ecc and clients) implemented by invoke
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// returns a set of credentials registered for a given chaincode id
+queryCCInfos(chaincode_id string) (Chaincode_EK []byte, allCredentials []Credentials)
```

Comment:
So what is the endorsing peer discovery protocol? I would imaging the following:

1. Client invokes `queryCCInfos`, and gets back the chaincode_ek and a list of enclave credentials. This list represents all possible endorsing enclaves/peers.

1. Client picks one (or more) enclaves using the information he got from credentials.

1. Last, clients prepare transaction proposal and submits it to the endorsing peers/enclaves.

With this protocol in mind, we would need to add some peer identifier to the enclave credentials, that allows the client to retrieve the peer endpoint. Also, since the credentials are attested, we have a nice way to show that an enclave is owned by a peer.

Reply:
A added `bytes peer_msp_id` to the credential message

Reply:
I forgot that for designated endorsing peer (rather than single peer), the client should know the correct peer unless she wants to broadcast the request (which works given the CC_EK).  That said, i don't really like the idea of adding msp_id to CC_Params for a number of reasons. (1) it is very designated endorsing peer specific and so i think an idea is better associated with the enclave key rather then CC_Pararms . (2) if you restrict it to MSP (rather than some form of peer id), it already exists in the endorsement policy of the cc, so really no need to replicate info?

So to me before changing anything in CC_Params and/or adding more discovery info to ERCC i think we should first answer the following question:  Is there a (org-independent/global) peer identity which (a) we can easily use to direct transactions, (b) is immutable and (c) can easily be retrieved either directly as part of create_enclave flows or by client to be additional input to create_enclave.

If there is none, we anyway have to broadcast so potentially you do need the enclave key as a client, only the cc_key. And if you want to narrow to orgs, you already can do that based on endorsement policy, again no need for additional discovery service from ercc?

Reply:
Ouch! this is a tricky thing. We indeed want the peer identity, which contains the peer msp_id + the peer public key. I added a host_params that contains the peer identity and is loaded with iniEnclave. Still need to figure out how we can pass the peer public key.